### PR TITLE
Upgrade to Zipkin 2.0

### LINKED ISSUE #26: Migrate to Zipkin 2.0
Zipkin PHP 2.0 is out and brought support for PHP 7.1+ avoiding type checks with almost no breaking changes. Consider upgrading.

### PR DESCRIPTION
**Pull request by adiachenko**
Closes #26 

Before merge it is advised to test this in a running Laravel app just in case something goes wrong.

### CODE REVIEW

**Comment by jcchavezs** [comment]
You can also introduce something like
https://github.com/jcchavezs/zipkin-instrumentation-symfony/tree/master/tests/E2E
that runs on CI.

lør. 17. okt. 2020 kl. 18:27 skrev Alexander Diachenko <
notifications@github.com>:

**Review by Mulkave — APPROVED** [review]

**Comment by Mulkave** [comment]
@KinaneD @IlliaBalia let's test this somewhere in our fleet to ensure it doesn't break anything as per @adiachenko 's comment.

**Comment by jcchavezs** [comment]
Friendly ping: Any movement on this one?

**Comment by KinaneD** [comment]
> Friendly ping: Any movement on this one?

Hi @jcchavezs will prioritize soon, sorry for the delay.

**Comment by KinaneD** [comment]
Hey guys, sorry for the delay again, i had some time to test the upgrade today on a couple of services, and i noticed a couple of "weird" behaviors.

_Zipking 2.*_
1. Not all commands are reported to Jeager.
   In the below screenshot we can find that only one of two commands has been traced. The other command and its corresponding feature `App/Console/Command/FetchDataAPIPerformanceInsights` & `App/Feature/FetchPerformanceFeature` were not, as opposed to `Zipkin 1.*`, when both commands implement `ShouldBeTraced`.
   Screen Shot 2020-11-20 at 2 27 17 PM
2. The title of the trace, in this case a feature, is gibberish.
Screen Shot 2020-11-20 at 2 21 42 PM

_Zipkin 1.*_
1. All commands are reported to Jeager.
   Although differently, one highlighting the path to the command, while the other the command itself, without the params
   though.
Screen Shot 2020-11-20 at 5 32 41 PM

2. The title of the trace, is clear and readable.
Screen Shot 2020-11-20 at 5 34 23 PM
Screen Shot 2020-11-20 at 5 33 02 PM

Any idea what could lead to the above? Were there any updates that perhaps we did  not account for?

**Comment by jcchavezs** [comment]
I recommend you download the trace from jaeger UI and paste it here for
both cases. I suspect this is something related to this class
https://github.com/openzipkin/zipkin-php/blob/master/src/Zipkin/Reporters/JsonV2Serializer.php
which was just introduced in zipkin PHP 2.0

**Comment by adiachenko** [comment]
@KinaneD Sorry for maybe asking the obvious, but I need to be sure. 

When sharing the results from Zipkin 1, did you test the latest released version of the package or a couple of versions back? Since we did Laravel 8 upgrade recently it may theoretically affect the accuracy of the test. Are there any differences between test subjects other than Zipkin 2? If not, I am baffled by the scenario description as well.

**Comment by jcchavezs** [comment]
@KinaneD could you please try with `openzipkin/zipkin:dev-fix_serialization_span_name`? I added a fix in https://github.com/openzipkin/zipkin-php/pull/189

**Comment by KinaneD** [comment]
> @KinaneD Sorry for maybe asking the obvious, but I need to be sure.
> 
> When sharing the results from Zipkin 1, did you test the latest released version of the package or a couple of versions back? Since we did Laravel 8 upgrade recently it may theoretically affect the accuracy of the test. Are there any differences between test subjects other than Zipkin 2? If not, I am baffled by the scenario description as well.

Hi @adiachenko yes v1.3.2

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "illuminate/queue": "~5.5.0|~5.6.0|~5.7.0|~5.8.0|^6.0|^7.0|^8.0",
         "illuminate/routing": "~5.5.0|~5.6.0|~5.7.0|~5.8.0|^6.0|^7.0|^8.0",
         "illuminate/support": "~5.5.0|~5.6.0|~5.7.0|~5.8.0|^6.0|^7.0|^8.0",
-        "openzipkin/zipkin": "~1.0",
+        "openzipkin/zipkin": "~2.0",
```
Comment (jcchavezs) [inline composer.json:24]:
```suggestion
        "openzipkin/zipkin": "~2.0.2",
```

This fixes @KinaneD's issue.

**Comment by jcchavezs** [comment]
@KinaneD I could reproduce my issue using https://github.com/openzipkin/zipkin-php-example so I cut a release to fix it: https://github.com/openzipkin/zipkin-php/releases/tag/2.0.2

**Comment by jcchavezs** [comment]
@adiachenko @KinaneD could we move forward on this?

**Comment by adiachenko** [comment]
Sure, we should be safe to merge this as long as we **release this a new major version** (which we technically should do anyway since we changed contract for extending propagation).

If new issues arise, I propose to address them separately in order to move this PR forward.

**Pull request merged**